File: client-android/serenada-core/src/main/java/app/serenada/core/SerenadaConfig.kt
package app.serenada.core

/**
 * Configuration for the Serenada SDK.
 */
data class SerenadaConfig(
    /** Server host or origin (e.g. "serenada.app" or "http://localhost:8080"). */
    val serverHost: String,
    /** Whether audio starts enabled (default true). */
    val defaultAudioEnabled: Boolean = true,
    /** Whether video starts enabled (default true). */
    val defaultVideoEnabled: Boolean = true,
    /** Enable experimental HD video capture. */
    val isHdVideoExperimentalEnabled: Boolean = false,
    /** Preferred signaling transports in priority order (default: WS then SSE). */
    val transports: List<SerenadaTransport> = listOf(SerenadaTransport.WS, SerenadaTransport.SSE),
)

/** Available signaling transport types. */
enum class SerenadaTransport {
    /** WebSocket transport. */
    WS,
    /** Server-Sent Events transport. */
    SSE,
}

File: client-ios/SerenadaCore/Sources/SerenadaConfig.swift
import Foundation

/// Signaling transport type.
public enum SerenadaTransport: String, Equatable, Sendable {
    case ws
    case sse
}

/// SDK configuration.
public struct SerenadaConfig: Equatable, Sendable {
    /// Server host or origin (e.g. "serenada.app" or "localhost:8080").
    public let serverHost: String
    /// Whether audio is enabled when joining a call. Defaults to `true`.
    public var defaultAudioEnabled: Bool
    /// Whether video is enabled when joining a call. Defaults to `true`.
    public var defaultVideoEnabled: Bool
    /// Preferred signaling transports in priority order. Defaults to `[.ws, .sse]`.
    public var transports: [SerenadaTransport]

    public init(
        serverHost: String,
        defaultAudioEnabled: Bool = true,
        defaultVideoEnabled: Bool = true,
        transports: [SerenadaTransport] = [.ws, .sse]
    ) {
        self.serverHost = serverHost
        self.defaultAudioEnabled = defaultAudioEnabled
        self.defaultVideoEnabled = defaultVideoEnabled
        self.transports = transports
    }
}
